src/pages/ArticleList.jsx

```diff
@@ -1,43 +1,62 @@
-import { useEffect, useState } from "react"
-import newsAPIGet from "../utils/utils"
-import ArticleCard from "../components/ArticleCard"
+import { useEffect, useState } from "react";
+import newsAPIGet from "../utils/utils";
+import ArticleCard from "../components/ArticleCard";
+import { useParams } from "react-router-dom";
 
-export default function ArticleList(){
-    const [articlesList, setArticlesList] = useState([])
-    const [isLoading, setIsLoading] = useState(false);
-    const [errorState, setErrorState] = useState(null);
+export default function ArticleList() {
+	const [articlesList, setArticlesList] = useState([]);
+	const [isLoading, setIsLoading] = useState(false);
+	const [errorState, setErrorState] = useState(null);
 
-    useEffect(()=>{
-        setIsLoading(true)
+	const { topicSlug } = useParams();
 
-        newsAPIGet("/articles")
-        .then(({data})=>{
-            const {articles} = data;
-            setIsLoading(false)
-            setArticlesList(articles)
-        })
-        .catch(()=>{
-            setIsLoading(false)
-            setErrorState(true)
-        })
+	useEffect(() => {
+		setIsLoading(true);
 
-    },[])
+		newsAPIGet("/articles")
+			.then(({ data }) => {
+				const { articles } = data;
+				setIsLoading(false);
+				if (topicSlug) {
+					const topicArticles = articles.filter((article) => {
```

Comment:
We should be making use of queries to achieve this filtering by topic rather than front end filtering. Our back end accepts a query of topic so lets make use of it 👍

Reply:
Thanks RIch - when I looked at my own backend docs I was surprised we didn't have this as an endpoint, so set this up instead; it didn't occur to me to use the API query! I'll refactor this in the next PR...